Building on ubuntu-24.04-arm

fyi, I tried to build master on ubuntu-24.04-arm (by a Github action) using Conda and our conda-forge/hpx-feedstock setup. It failed with the one below, which I could solve by adding `-D HPX_WITH_GENERIC_CONTEXT_COROUTINES=On` to CMake and then the build succeeds :)
I get the impression that setting this option should not be required but detected and added automatically (?), so I just share my experiences

```
...s/hpx_coroutines.dir/src/detail/context_base.cpp.o.d -o libs/core/coroutines/CMakeFiles/hpx_coroutines.dir/src/detail/context_base.cpp.o -c $SRC_DIR/libs/core/coroutines/src/detail/context_base.cpp
 │ │ In file included from $SRC_DIR/libs/core/coroutines/include/hpx/coroutines/detail/context_impl.hpp:126,
 │ │                  from $SRC_DIR/libs/core/coroutines/include/hpx/coroutines/detail/context_base.hpp:36,
 │ │                  from $SRC_DIR/libs/core/coroutines/src/detail/context_base.cpp:12:
 │ │ $SRC_DIR/libs/core/coroutines/include/hpx/coroutines/detail/context_linux_x86.hpp: In member function 'void hpx::threads::coroutines::detail::lx::x86_linux_context_impl_base::prefetch() const':
 │ │ $SRC_DIR/libs/core/coroutines/include/hpx/coroutines/detail/context_linux_x86.hpp:206:41: error: static assertion failed
 │ │   206 |             static_assert(sizeof(void*) == 4);
 │ │       |                           ~~~~~~~~~~~~~~^~~~
 │ │ $SRC_DIR/libs/core/coroutines/include/hpx/coroutines/detail/context_linux_x86.hpp:206:41: note: the comparison reduces to '(8 == 4)'
 │ │ ninja: build stopped: subcommand failed.
 │ │ × error Script failed with status 1
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1908,10 +1908,15 @@ endif()
 if("${HPX_PLATFORM_UC}" STREQUAL "BLUEGENEQ")
   set(__use_generic_coroutine_context ON)
 endif()
-# If compiling for riscv64, arm or arm64 (including armv5, armv6, armv7), automatically bake in Boost.Context
-if(${__target_arch} STREQUAL "riscv64" OR ${__target_arch} STREQUAL "arm64"
-  OR ${__target_arch} STREQUAL "arm" OR ${__target_arch} STREQUAL "armv5"
-  OR ${__target_arch} STREQUAL "armv6" OR ${__target_arch} STREQUAL "armv7")
+# If compiling for riscv64, arm or arm64 (including armv5, armv6, armv7),
+# automatically bake in Boost.Context
+if(${__target_arch} STREQUAL "riscv64"
+   OR ${__target_arch} STREQUAL "arm64"
+   OR ${__target_arch} STREQUAL "arm"
+   OR ${__target_arch} STREQUAL "armv5"
+   OR ${__target_arch} STREQUAL "armv6"
+   OR ${__target_arch} STREQUAL "armv7"
+)
   set(__use_generic_coroutine_context ON)
 endif()
 

